### PR TITLE
Update to new deployment

### PR DESCRIPTION
**Pull request by Smaug123**
@jgiannuzzi Could you confirm that this matches the details you had? I believe this should work, but GitHub Actions are a bit of a dark art.

----

I believe reviewers can't see the secrets available in [the new deployment](https://github.com/G-Research/ApiSurface/deployments/release).

### CODE REVIEW

File: .github/workflows/dotnet.yaml [hunk 1]
```diff
@@ -159,14 +159,20 @@ jobs:
     runs-on: ubuntu-latest
     if: ${{ !github.event.repository.fork && github.ref == 'refs/heads/main' }}
     needs: [all-required-checks-complete]
-    environment: main-deploy
-    permissions:
-      contents: write
+    environment: release
```
Comment (Smaug123) [inline .github/workflows/dotnet.yaml:162]: Note the change of environment. The new environment has creds for a GitHub bot in.

File: .github/workflows/dotnet.yaml [hunk 2]
```diff
@@ -159,14 +159,20 @@ jobs:
     runs-on: ubuntu-latest
     if: ${{ !github.event.repository.fork && github.ref == 'refs/heads/main' }}
     needs: [all-required-checks-complete]
-    environment: main-deploy
-    permissions:
```
[the remark this answers is not in the document]
Comment (Smaug123) [reply .github/workflows/dotnet.yaml:163]: From [the docs](https://docs.github.com/en/rest/apps/apps?apiVersion=2022-11-28#create-an-installation-access-token-for-an-app):

> If permissions is not specified, the installation access token will have all of the permissions that were granted to the app. 

So if everything's behaving as documented, then indeed we shouldn't need to do anything special to get write permission.

[the remark this answers is not in the document]
Comment (jgiannuzzi) [reply .github/workflows/dotnet.yaml:163]: Correct

**Review by jgiannuzzi — APPROVED** [review]
I confirm that the secrets you used are the ones we created with the app credentials yesterday. Before you merge, let me check whether the app has the permissions to create a release though! (It only has `contents:write` permissions at the moment)

**Comment by jgiannuzzi** [comment]
I confirm that `contents:write` is enough to create releases.

**Comment by Smaug123** [comment]
Hold onto your hats!

**Pull request merged**